ci/dockerfiles/rc/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM ubuntu:bionic
+
+
+ENV \
+  CONCOURSE_RESOURCE_TYPES=/concourse-resource-types \
+  CONCOURSE_WORK_DIR=/worker-state \
+  CONCOURSE_GARDEN_DNS_PROXY_ENABLE=true
+
+
+RUN apt update && apt install -y \
+    ca-certificates \
+    btrfs-tools \
+    dumb-init \
+    iproute2
+
+
+ADD ./rc/bin/concourse /usr/local/bin
```

Comment:
I think you can just do this:

```
ADD ./rc/concourse-*.tgz /usr/local
```

...which would remove the need to extract it in the task (`ADD` extracts local archives). This should result in `/usr/local/concourse/{bin,fly-assets,...}`.

The advantage of this is that it'll match the install instructions for the tarball distribution, and when installed in this way the binary can automatically find the `fly-assets`, `resource-types`, `bin/gdn` etc.

This is how the smoke tests install, too: 

https://github.com/concourse/concourse/blob/5dd0bcc92d24858af0830c27ad3c3d652513f8bb/ci/deployments/smoke/smoke.tf#L143

This install method is partly inspired by Go (which typically installs to `/usr/local/go`).

Reply:
Neat! Thanks!

Reply:
Weird, I was trying the `ADD` with wildcards here, and it seems like it doesn't extract - if I try it without wildcards though (passing the full name), it works just fine.

Looking at the buildkit test suite for the Dockerfile frontend (see https://github.com/tonistiigi/buildkit/blob/6cb4256e81dd20691d9c71a988d0791aee02e00b/frontend/dockerfile/dockerfile_test.go#L38-L49), it appears that they indeed don't have a test for such case. 

<details>
<summary>with wildcard</summary>
<pre>
docker build -t a .
[+] Building 2.9s (9/9) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                   0.0s
 => => transferring dockerfile: 1.09kB                                                                                                                 0.0s
 => [internal] load .dockerignore                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                        0.0s
 => [internal] load metadata for docker.io/library/ubuntu:bionic                                                                                       0.0s
 => CACHED [internal] helper image for file operations                                                                                                 0.0s
 => [1/3] FROM docker.io/library/ubuntu:bionic                                                                                                         0.0s
 => [internal] load build context                                                                                                                      0.0s
 => => transferring context: 1.41kB                                                                                                                    0.0s
 => CACHED [2/3] RUN apt update && apt install -y     btrfs-tools     ca-certificates     dumb-init     iproute2                                       0.0s
 => [3/3] ADD ./rc/*.tgz /usr/local                                                                                                                    1.9s
 => exporting to image                                                                                                                                 0.9s
 => => exporting layers                                                                                                                                0.9s
 => => writing image sha256:e0edec6723849d4aa97627de886d8fdfbcf43638b5001317a6e20a8c52dce258                                                           0.0s
 => => naming to docker.io/library/a                                                                                                                   0.0s

docker run --entrypoint /bin/sh a -c 'ls /usr/local | grep concourse'
concourse-5.0.0-rc.20-linux-amd64.tgz
</pre>
</details>

<details>
<summary>without wildcard</summary>
<pre>
docker build -t a .
[+] Building 0.1s (9/9) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                   0.0s
 => => transferring dockerfile: 1.12kB                                                                                                                 0.0s
 => [internal] load .dockerignore                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                        0.0s
 => [internal] load metadata for docker.io/library/ubuntu:bionic                                                                                       0.0s
 => [1/3] FROM docker.io/library/ubuntu:bionic                                                                                                         0.0s
 => [internal] helper image for file operations                                                                                                        0.0s
 => [internal] load build context                                                                                                                      0.0s
 => => transferring context: 1.41kB                                                                                                                    0.0s
 => CACHED [2/3] RUN apt update && apt install -y     btrfs-tools     ca-certificates     dumb-init     iproute2                                       0.0s
 => CACHED [3/3] ADD ./rc/concourse-5.0.0-rc.20-linux-amd64.tgz /usr/local                                                                             0.0s
 => exporting to image                                                                                                                                 0.0s
 => => exporting layers                                                                                                                                0.0s
 => => writing image sha256:84d4de2b138faf696517919c50dc4680c6584b0b75fa52e98daf41b73fe51a4f                                                           0.0s
 => => naming to docker.io/library/a                                                                                                                   0.0s

docker run --entrypoint /bin/sh a -c 'ls /usr/local | grep concourse'
concourse
</pre>
</details>

Pretty crazy

Reply:
Now, with "regular docker":

<details>
<summary>with wildcards but without buildkit</summary>
<pre>
docker build -t a .
Sending build context to Docker daemon  189.9MB
Step 1/15 : FROM ubuntu:bionic
 ---> 1d9c17228a9e
Step 2/15 : ENV CONCOURSE_SESSION_SIGNING_KEY     /concourse-keys/session_signing_key
 ---> Using cache
 ---> 2b83f8901a32
Step 3/15 : ENV CONCOURSE_TSA_AUTHORIZED_KEYS     /concourse-keys/authorized_worker_keys
 ---> Using cache
 ---> 9a9c5bb963ac
Step 4/15 : ENV CONCOURSE_TSA_HOST_KEY            /concourse-keys/tsa_host_key
 ---> Using cache
 ---> d375a751074d
Step 5/15 : ENV CONCOURSE_RESOURCE_TYPES          /concourse-resource-types
 ---> Using cache
 ---> 3f29985eb47c
Step 6/15 : ENV CONCOURSE_TSA_PUBLIC_KEY          /concourse-keys/tsa_host_key.pub
 ---> Using cache
 ---> 3d8b26416bc8
Step 7/15 : ENV CONCOURSE_TSA_WORKER_PRIVATE_KEY  /concourse-keys/worker_key
 ---> Using cache
 ---> c3b605c29336
Step 8/15 : ENV CONCOURSE_GARDEN_DNS_PROXY_ENABLE true
 ---> Using cache
 ---> d7cbd9ddc87e
Step 9/15 : ENV CONCOURSE_WORKER_GARDEN_DNS_PROXY_ENABLE true
 ---> Using cache
 ---> 892f1397ed33
Step 10/15 : ENV CONCOURSE_WORK_DIR                /worker-state
 ---> Using cache
 ---> f4009a1d1ded
Step 11/15 : ENV CONCOURSE_WORKER_WORK_DIR         /worker-state
 ---> Using cache
 ---> ad87b391b231
Step 12/15 : VOLUME /worker-state
 ---> Using cache
 ---> 54a045c252a0
Step 13/15 : RUN apt update && apt install -y     btrfs-tools     ca-certificates     dumb-init     iproute2
 ---> Using cache
 ---> 702389661ca0
Step 14/15 : ADD ./rc/*.tgz /usr/local
 ---> c96116a680cc
Step 15/15 : ENTRYPOINT ["dumb-init", "/usr/local/bin/concourse"]
 ---> Running in d3f0a65d232f
Removing intermediate container d3f0a65d232f
 ---> 016d336bd83c
Successfully built 016d336bd83c
Successfully tagged a:latest

docker run --entrypoint /bin/sh a -c 'ls /usr/local | grep concourse'
concourse
</pre>
</details>

🤷‍♂️

Reply:
That's annoying! I guess as a workaround we could just rename the file to a stable name like `concourse.tgz` so we don't need to wildcard in the Dockerfile.

Reply:
Ah, before I saw your comment, I had a patch locally using multi-stage builds, which end up achieving the goal in both cases (regular docker and buildkit). 

Do you think that's ok to keep?

thx!

Reply:
That sounds better, it'd be nice to just re-use the `builder` task without modifications.